### PR TITLE
Don't overwrite a provided literal default client

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

This fixes the issue we had today

/assign @fejta @cjwagner 
/cc @BenTheElder @krzyzacy

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
We could unit test this func but ideally we want to test kubeconfig file content + cluster yaml file content --> correct contexts and aliases

**Comment by fejta** [comment]
> test kubeconfig file content + cluster yaml file content --> correct contexts and aliases

This seems like the right test. Don't see anything preventing this being a unit test (two strings as input results in the expected map)?

**Comment by stevekuznetsov** [comment]
The hard thing here is I would love to validate that `kube.config` and `build-cluster.yaml` come together to make the correct clients at the correct aliases, but the output types of the helpers here are upstream interfaces and have no data about the inputs that were used to make them. I'll try wrapping them so they're testable.

**Comment by stevekuznetsov** [comment]
@fejta @cjwagner updated to sanity-check needed aliases versus loaded ones, added some unit testing

Majority of unit tests are here: https://github.com/kubernetes/test-infra/pull/11227

File: prow/flagutil/kubernetes_cluster_clients.go [hunk 1]
```diff
@@ -174,16 +175,33 @@ func (o *KubernetesOptions) BuildClusterClients(namespace string, dryRun bool) (
...
-		if context == o.infraContext {
-			context = kube.DefaultClusterAlias
-		}
-		buildClients[context] = client.CoreV1().Pods(namespace)
+	for alias, client := range contextsToAliases(o.kubernetesClientsByContext, o.infraContext) {
+		buildClusterAliases.Insert(alias)
+		buildClients[alias] = client.CoreV1().Pods(namespace)
+	}
+	if !buildClusterAliases.HasAll(knownAliases.UnsortedList()...) {
+		return nil, fmt.Errorf("the following cluster aliases declared for jobs were not found in loaded clusters: %v", knownAliases.Difference(buildClusterAliases).List())
```
Comment (stevekuznetsov) [inline prow/flagutil/kubernetes_cluster_clients.go]: Almost makes me think we should do this on config-load as well ... or mount the cluster credentials into the `checkconfig` job and do it pre-merge as well. @cjwagner WDYT?

**Comment by krzyzacy** [comment]
looks sane - assume there's no functional changes in https://github.com/kubernetes/test-infra/pull/11227 and if we take in this PR we should be back to running state?

**Comment by stevekuznetsov** [comment]
The other PR errors on otherwise undocumented edge cases, but in the normal paths should be a no-op yes.
